### PR TITLE
[10.0][FIX] account_invoice_inter_company: payment_term_id and reference

### PR DESCRIPTION
**Pull request by chafique-delli**

### CODE REVIEW

**Review by mourad-ehm — APPROVED** [review]
Code reviw

File: account_invoice_inter_company/models/account_invoice.py [hunk 1]
```diff
@@ -173,13 +173,15 @@ def _prepare_invoice_data(self, dest_inv_type, dest_journal_type,
         }
         dest_partner_data = self.env['account.invoice'].play_onchanges(
             dest_partner_data, ['partner_id'])
+        if not dest_partner_data.get('payment_term_id'):
+            dest_partner_data['payment_term_id'] = self.payment_term_id.id
```
Comment (pedrobaeza) [inline account_invoice_inter_company/models/account_invoice.py]: Payment terms might not shared between companies, so this doesn't serve as is.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
And you have to change module version number

File: account_invoice_inter_company/models/account_invoice.py (hunk 1 above)
Comment (chafique-delli) [reply account_invoice_inter_company/models/account_invoice.py]: @pedrobaeza , You're right, I was too fast in my PR.
I'm sorry, I correct.

Comment (chafique-delli) [reply account_invoice_inter_company/models/account_invoice.py]: @pedrobaeza , I wanted to have your opinion to share the terms of payment. I thought to add a boolean field on account.payment.term to say which one wants to share. And with an access rule, I authorize those who have this boolean checked ?

Comment (pedrobaeza) [reply account_invoice_inter_company/models/account_invoice.py]: In the spirit on inheritability, I think you can test if there are read permissions for the target company, and if affirmative, then assign it. This way, you can implement any strategy for sharing them (or use the default of not assigning company).

Comment (chafique-delli) [reply account_invoice_inter_company/models/account_invoice.py]: @pedrobaeza , ok, thank you.

Comment (alexis-via) [reply account_invoice_inter_company/models/account_invoice.py]: I agree with Pedro. It's surprising that there are no record rules by default on payment term, which is a bit surprising. I made a PR to add it to see how the editor considers it https://github.com/odoo/odoo/pull/23607 But checking for read perm in the target company via the generic method for that and then add it to destination invoice or leave the field empty. If the field payment_term_id is left empty, then the due_date should be copied.

File: account_invoice_inter_company/models/account_invoice.py [hunk 2]
```diff
@@ -173,13 +173,15 @@ def _prepare_invoice_data(self, dest_inv_type, dest_journal_type,
...
         return {
             'name': self.name,
             'origin': _('%s - Invoice: %s') % (self.company_id.name,
                                                self.number),
             'type': dest_inv_type,
             'date_invoice': self.date_invoice,
-            'reference': self.reference,
+            'reference': _('Invoice: %s') % self.number,
```
Comment (alexis-via) [inline account_invoice_inter_company/models/account_invoice.py]: I think we should just put "self.number" in that field, without 'Invoice' prefix.

**Comment by chafique-delli** [comment]
@pedrobaeza , Hi, I made the changes after your comments and those of @alexis-via , can you tell me what you think. Thank you.

**Comment by chafique-delli** [comment]
@alexis-via , Hi, I made the changes after your comments and those of @pedrobaeza , can you tell me what you think. Thank you.

File: account_invoice_inter_company/models/account_invoice.py [hunk 3]
```diff
@@ -173,13 +173,20 @@ def _prepare_invoice_data(self, dest_inv_type, dest_journal_type,
...
+                    dest_user).check_access_rule('read'):
+                dest_partner_data['payment_term_id'] = self.payment_term_id.id
         return {
             'name': self.name,
             'origin': _('%s - Invoice: %s') % (self.company_id.name,
                                                self.number),
             'type': dest_inv_type,
             'date_invoice': self.date_invoice,
-            'reference': self.reference,
+            'reference': self.number,
```
Comment (pedrobaeza) [inline account_invoice_inter_company/models/account_invoice.py:189]: Why this change?

Comment (chafique-delli) [reply account_invoice_inter_company/models/account_invoice.py:189]: @pedrobaeza ,I think it's clearer to have the original invoice number in the 'supplier reference' field.
I know it can duplicate with the field 'origin'.
By the way, the field 'origin', I could leave it empty.

Comment (pedrobaeza) [reply account_invoice_inter_company/models/account_invoice.py:189]: OK, thanks for the clarification. Indeed it's a good option. Leave both.

**Review by pedrobaeza — APPROVED** [review]
